src/GamingDevice.js

```diff
@@ -0,0 +1,77 @@
+let gamingDeviceId = 0;
+const defaultNumOfButtons = 18;
+
+import { GamingDeviceButton } from "./GamingDeviceButton.js";
+
+export class GamingDevice extends HTMLDivElement {
+  constructor(numOfButtons = defaultNumOfButtons) {
+    super();
+
+    const id = gamingDeviceId++;
+
+    // setup the gamepad's front end
+    this.innerHTML = `
+      <style>
+        .gaming-device {
+          border: 1px solid grey;
+          display: inline-block;
+          margin: 5px;
+        }
+        .connected {
+          background-color: lightblue;
+        }
+      </style>
+      Hello gamepad ${id}!
+    `;
+    this.classList.add("gaming-device");
+
+    // set default attributes
+    this._id = id;
+    this._connected = false;
+    this._buttons = gamingDeviceButtons();
+
+    addButtonsToButtonDiv(this);
+
+    function gamingDeviceButtons() {
+      const buttons = Array(numOfButtons)
```

Comment:
Note that calling `Array()` already fills the array with empty slots, so .fill() is redundant. What would have been good here is maybe. I'd suggest just doing a regular for loop here.